vassal-app/src/main/java/VASSAL/counters/FreeRotator.java

```diff
@@ -227,7 +227,7 @@ public boolean isFreeRotation() {
 
   public void setAngle(double angle) {
     if (isFreeRotation()) {
-      validAngles[angleIndex] = angle;
+      validAngles[angleIndex] = ((angle % 360) - 360) % 360;
```

Comment:
Rather than repeating the angle normalization, it would be better to move the original lines 235-238 out ahead of the if block:
```
      // We (stupidly) store allowed angles in descending order from 0.
      // Normalize the angle to be in (-360, 0] to match that.
      angle = ((angle % 360) - 360) % 360;

      if (isFreeRotation()) { 
```